pandas/core/dtypes/cast.py

```diff
@@ -680,7 +686,8 @@ def maybe_promote(dtype, fill_value=np.nan):
             fill_value = dtype.type("NaT", "ns")
         else:
             dtype = np.dtype(np.object_)
-            fill_value = np.nan
+            if fill_value is not libmissing.NA:
```

Comment:
does ```fill_value or np.nan``` work generally? (or missing an attribute on libmissing.NA?

Reply:
pretty sure that would raise when fill_value is pd.NA.

Reply:
if this is true then its a bug e.g.
```
In [189]: 1 or np.nan                                                                                                                                                                     
Out[189]: 1
```

as this should work with pd.NA in an or clause

Reply:
> if this is true then its a bug e.g.

I don't know if its technically a bug bc `NA.__bool__` raises intentionally, but it _is_ a PITA that will continue biting us. 

```
>>> pd.NA or 1
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "pandas/_libs/missing.pyx", line 360, in pandas._libs.missing.NAType.__bool__
    raise TypeError("boolean value of NA is ambiguous")
TypeError: boolean value of NA is ambiguous
```

Reply:
ok do we have an issue for this, I think we need to change this. (if not can you create).

Reply:
cc @jorisvandenbossche @TomAugspurger

Reply:
This is done on purpose, yes

Reply:
then we need to revert it. it doesn't make sense. what is rational here? this is *different* than np.nan and AFAICT doesn't offer any actual advanatages.

Reply:
`bool(pd.NA)` raising is a pain, but there isn't a right answer that's consistent with the definition of `pd.NA`: we don't know if the actual value of the `pd.NA` is truthy or falsey, so raising is the least-worst option.

Reply:
hmm, what advantage does this convey? IOW where is this actually useful to do.

Reply:
The `bool(pd.NA)` behaviour was certainly not the most discussed aspect in the original issues/PRs about introducing pd.NA, so it might be good to have a more detailed discussion about it. But we should then open a dedicated issue about it. 

(and as Tom notes, I don't think there is a clear "best" solution)